### DISCUSSION TITLE
Usage of Backstage apiRef

### DISCUSSION THREAD

**Discussion by gaelgoth** [procore-oss/backstage-plugin-announcements#61] [Q&A] [answered]
Hello,

We recently used your (archived) [plugin fork](https://github.com/kurtaking/backstage-plugin-announcements). Since I would like to migrate to the *procore-oss* organization version, you have introduced the usage of the Backstage API, as referenced in [this commit](https://github.com/procore-oss/backstage-plugin-announcements/commit/73a7e8e011a687375eeb0c5fcf436ac276b658f6). However, I have not been able to pass `fetchApi` in my `search.ts`. Do you have an example of how to do it properly? I would be happy to contribute and help by updating the documentation 😄

```typescript
...
import { AnnouncementCollatorFactory } from '@procore-oss/backstage-plugin-announcements-backend';
import { PgSearchEngine } from '@backstage/plugin-search-backend-module-pg';
import { Router } from 'express';

export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
  const searchEngine = await PgSearchEngine.fromConfig(env.config, { database: env.database });
  const indexBuilder = new IndexBuilder({
    logger: env.logger,
    searchEngine,
  });

...

  // Announcements indexing
  indexBuilder.addCollator({
    schedule: schedule,
    factory: AnnouncementCollatorFactory.fromConfig({
      logger: env.logger,
      discoveryApi: env.discovery,
      fetchApi: env.fetchApi, // FIXME: Correctly pass fetchApi
    }),
  });

  const { scheduler } = await indexBuilder.build();
  scheduler.start();

  useHotCleanup(module, () => scheduler.stop());

  return await createRouter({
    engine: indexBuilder.getSearchEngine(),
    types: indexBuilder.getDocumentTypes(),
    permissions: env.permissions,
    config: env.config,
    logger: env.logger,
  });
}

```

**Comment by kurtaking** [comment, marked as answer]
Hey @gaelgoth, 

I've searched for other examples and confirmed this with a few others. I should not have moved this to use the `FetchApi` as it's a front-end concept. The best option is to revert to using (native) fetch, which I can take care of in the next few hours unless you see this and want to contribute.

**Reply by kurtaking** [reply]
@gaelgoth, please upgrade to v0.3.1 and let me know how your experience goes.

**Reply by kurtaking** [reply]
Upgrading to the latest versions should be a 1:1 migration from the original plugins. Let me know how things go!

**Reply by gaelgoth** [reply]
Hello @kurtaking ,

Thank you for the quick fix! Everything works as expected👍🏾
